### ISSUE TITLE
it works very well (bootstrap-select-v1.14.0-beta3-updated and bootstrap 5.3.0)

### ISSUE THREAD

**Issue by mattymatty76** [snapappointments/bootstrap-select#2837]
I have modified the file bootstrap-select-v1.14.0-beta3
and looks like working very well with bootstrap 5.3.0, I published the source code here 

[https://gist.github.com/mattymatty76/c996d3b77f298b2ec133be59992df9d4](https://gist.github.com/mattymatty76/c996d3b77f298b2ec133be59992df9d4)

I hope this is helpful

Thank you

**Comment by XusBadia** [comment]
Thanks! I am really trying to get it to work with Bootstrap 5 as well as it did in v4, since it's the best dropdown select plugin out there. I am getting trouble getting the list to build on page load (it only builds after clicking on the dropdown once). Have you managed to get that working?

Thanks for keeping this project alive.

**Comment by mattymatty76** [comment]
You are welcome
it's the same behavior that you find in the previous version as well.
When you open for the first time the page, that has a select, until you click on select you don't have li tags in .dropdown-menu
But you can bypass it putting this code after loading page...

```
<script>
$(document).ready(function() {
    $('.selectpicker').selectpicker('toggle');
});
</script>
```

**Comment by XusBadia** [comment]
I've looked everywhere for that but couldn't find it! Thanks!

Is there a way to toggle and have the li tags, but prevent it from actually open the dropdown?

Thanks @mattymatty76 for your help!

**Comment by mattymatty76** [comment]
Yes you can toggle two times, and the select will open and close immediately its dropdown-menu...
And in this manner the library bootstrap-select draws the list that you want with li tags

**Comment by JMACSilvestre** [comment]
Hi there! Could you help me to implement bootstrap-select in vite laravel? I have tried with bootstarp 5.3 and the original plugin 1.14.0-beta3. If o use the one you modified it Will work? Thanks!

**Comment by joshuap** [comment]
@mattymatty76 it appears that snippet.host is offline. Can you share the code as a GitHub gist?

**Comment by jorbs** [comment]
@mattymatty76 are you able to render the placeholder text properly?

**Comment by mattymatty76** [comment]
@joshuap 

you can try with this, now the link works

[https://gist.github.com/mattymatty76/c996d3b77f298b2ec133be59992df9d4](https://gist.github.com/mattymatty76/c996d3b77f298b2ec133be59992df9d4)

about that code in GitHub, I should create a fork of this project and put the library that I have modified in that link... but I don't know if it can help or confuse people

**Comment by mattymatty76** [comment]
@JMACSilvestre 
> Hi there! Could you help me to implement bootstrap-select in vite laravel? I have tried with bootstarp 5.3 and the original plugin 1.14.0-beta3. If o use the one you modified it Will work? Thanks!

maybe I don't know but you can try...

[https://gist.github.com/mattymatty76/c996d3b77f298b2ec133be59992df9d4](https://gist.github.com/mattymatty76/c996d3b77f298b2ec133be59992df9d4)

**Comment by g-pane** [comment]
@mattymatty76 make a fork for this. I have a problem, also with the original one. The "required" HTML5 tag not showing the message if the select is inside a "row" of Bootstrap 5. Can you solve it? Thanks

**Comment by mattymatty76** [comment]
Hi @g-pane,
could you give me an example please: I don'tknow maybe in jsfiddle or codepen...

Thank you, m.

**Comment by g-pane** [comment]
`<div class="row">
            <div class="col-xl-12 mx-auto mt-3">
                <div class="card">
                    <div class="card-body">
                        <div class="row">
                            <div class="col-6">
                                <h5 class="mb-0">TEST</h5>
                            </div>
                            <div class="col-6 text-end">
                                <h5 class="mb-0 d-inline">TEST: </h5><h5 id="test" class="mb-0 d-inline bg-danger rounded text-white p-1">TEST2</h5>
                                <h6 class="mb-0 d-inline">- UM:</h6><h6 id="um" class="mb-0 d-inline p-1">?</h6>
                            </div>
                        </div>
                        <hr/>
                        <form>
                            <div class="row gy-3 px-4 py-3 row-cols-1 row-cols-sm-2 row-cols-md-4 row-cols-lg-8">
                                <div class="col">
                                    <h6 class="mb-0">SELECT</h6>
                                    <select id="test2" name="test2"
                                            class="form-control selectpicker mt-2"
                                            data-live-search="true" title="Test2" required>
                                    </select>
                                </div>`

If i remove "row gy-3 px-4 py-3 row-cols-1 row-cols-sm-2 row-cols-md-4 row-cols-lg-8" i see the message again.

**Comment by mattymatty76** [comment]
try this now it works

```
<div class="row"> 
    <div class="col-xl-12 mx-auto mt-3">
        <div class="card"> 
            <div class="card-body">
                <div class="row">
                    <div class="col-6"><h5 class="mb-0">TEST</h5></div>
                    <div class="col-6 text-end"> 
                        <h5 class="mb-0 d-inline">TEST: </h5>
                        <h5 id="test" class="mb-0 d-inline bg-danger rounded text-white p-1">TEST2</h5>
                        <h6 class="mb-0 d-inline">- UM:</h6>
                        <h6 id="um" class="mb-0 d-inline p-1">?</h6>
                    </div>
                </div>
                <hr>
                <form>
                    <div class="row gy-3 px-4 py-3 row-cols-1 row-cols-sm-2 row-cols-md-4 row-cols-lg-8">
                        <div class="col">
                            <h6 class="mb-0">SELECT</h6>
                            <select id="test2" name="test2" class="form-control selectpicker mt-2" data-live-search="true" title="Test2" required>
                                <option value="1">one</option>
                                <option value="2">two</option>
                            </select>
                        </div>
                    </div>
                </form>
            </div>
        </div>
    </div>
</div>
```

**Comment by g-pane** [comment]
Not for me, i don't see the message. Only the focus on the button (p.s. submit button missing from the code, just for give you an idea)

**Comment by mattymatty76** [comment]
![works](https://github.com/snapappointments/bootstrap-select/assets/56148940/26687cce-9d8a-4818-9e53-56f7b9db99c9)

in my case works

**Comment by mattymatty76** [comment]
you can bypass the issue trying a jquery.validate.js library and redefine more or less this code to adapt to your case...

```
 $("form").validate({
		onfocusin: function(e) {this.element(e);},
		errorPlacement: function(error, element) {
			error.insertAfter(element);
		},
		highlight: function(element, errorClass, validClass) {
			if ($(element).is("select") ){
				if ($(element).parent().hasClass('bootstrap-select')) {
					$(element).parent().addClass(errorClass).removeClass(validClass);
				}else $(element).addClass(errorClass).removeClass(validClass);
			}else{
				$(element).addClass(errorClass).removeClass(validClass);
			}
		},
	    unhighlight: function(element, errorClass, validClass) {
	    	if ($(element).is("select") ){
				if ($(element).parent().hasClass('bootstrap-select')) {
					$(element).parent().removeClass(errorClass).addClass(validClass);
				}else $(element).removeClass(errorClass).addClass(validClass);
			}else{
				$(element).removeClass(errorClass).addClass(validClass);
			}
	    }
	});
```

**Comment by mattymatty76** [comment]
this example works, I have just tested it

[https://jsfiddle.net/mattymatty76/ecd6azby/2/](https://jsfiddle.net/mattymatty76/ecd6azby/2/)

**Comment by dalai-lamia** [comment]
There is a problem that also existed in the beta-2 and beta-3 versions: when the selectpicker is in n input-group, it does not respect this class but instead displays on a new line, below the label.

Any ideas how to fix this?

**Comment by Moxh3** [comment]
Confirming this version works better with Bootstrap 5.3.0

**Comment by marcapas01** [comment]
> I have modified the file bootstrap-select-v1.14.0-beta3 and looks like working very well with bootstrap 5.3.0, I published the source code here
> 
> https://gist.github.com/mattymatty76/c996d3b77f298b2ec133be59992df9d4
> 
> I hope this is helpful
> 
> Thank you

Hello I got a error mesage Dropdown is not a a constructor. Do you have any solution to fix that? Thank you

**Comment by krishnarajsj** [comment]
could anyone provide me the corresponding bootstrap-select css?

Im facing below issue.

Wen I click on dropdown , its coming very close to the left margin and after a microsecond it alligns correctly

**Comment by nkavian** [comment]
Has anyone gotten this to work with Bootstrap's floating select? https://getbootstrap.com/docs/5.3/forms/floating-labels/#selects

**Comment by mattymatty76** [comment]
Hi @krishnarajsj,
you can use bootstrap-select.css v1.13.18 with or without adjustments or with your customizations...

**Comment by hasanmk52** [comment]
This library works well to most extent but when I am using it as a multi-select with optgroups then it is not toggling the selection and not showing the check-mark when I select. Have posted the screenshot where I am selecting two values but only one shows as selected along with the html generated where we see the `selected` class is not applied to the `li` and `a` tag

<img width="293" alt="Image" src="https://github.com/user-attachments/assets/63a07711-df43-48ba-b018-75a1680b3677" />

```
<li class="optgroup-1 selected">
<a role="option" class="dropdown-item opt selected" id="bs-select-4-3" tabindex="0" aria-selected="true" aria-setsize="27" aria-posinset="1">
<span class=" bs-ok-default check-mark"></span><span class="text">Premium</span>
</a>
</li>

<li class="optgroup-1">
<a role="option" class="dropdown-item opt" id="bs-select-4-4" tabindex="0" aria-selected="false" aria-setsize="27" aria-posinset="undefined">
<span class=" bs-ok-default check-mark"></span><span class="text">Compact</span>
</a>
</li>
```

**Comment by hasanmk52** [comment]
Hi, thanks for providing the updated library for BS 5. It works ok however I do face a few issues:

If the dropdown has large number of values then on click of the dropdown it takes time to open the list of values probably because it is trying to build the entire li element list at that time, so can we preinitialize that ? 



Another is when I use it as a multi-select with optgroups and when I click on more than one value the tick mark doesn't come up consistently as shown in below screenshot
<img width="591" alt="Image" src="https://github.com/user-attachments/assets/45dca4a9-6e31-45f8-b0bc-eac253f268ae" />

**Comment by mattymatty76** [comment]
Hi @hasanmk52,
have you tried this css file?  [https://gist.github.com/mattymatty76/43cb17bd11f76451cec2e2ff14fafbb2](https://gist.github.com/mattymatty76/43cb17bd11f76451cec2e2ff14fafbb2) 
it could be a problem related to the css style... In my case I have simulated your multi-menu without problems, so if you use bootstrap-select(v1.14.0-gamma1).js the only thing different for me is the css file...

regarding the slowness when you open menu, if the items are a lot you should use ajax select because there is a physical limit in loading elements due to memory saturation...

but before doing anything else, try to set this param in bootstrap-select(v1.14.0-gamma1).js 
virtualScroll: 1000000,
It might help in rendering

**Comment by hasanmk52** [comment]
Yes I did use the same CSS file and have also tried adjusting the virtualScroll and still facing both the issues. As for the slowness I have around 50k elements and when I click the dropdown it takes up-to 2 secs to open it.

**Comment by mattymatty76** [comment]
@hasanmk52 
you could try to build the menu, just to do a test, as key / value an incremental integer in key; this is because I had noticed that if the key was numeric up to 100k it did not give problems in opening (that is, it is acceptable as speed), but if key / value were both strings, then it slowed me down a lot: but these are all empirical tests, and in any case at those levels from 50k and up I would use ajaxselect.

In addition you can also try to trick it in this way: as soon as you create it, simulate in javascript a trigger that instantly opens and closes the menu: this should act as a sort of initialization of the menu rendering...

For the css try to do a bit of debugging because in my opinion you could have some css overlap: I tried to simulate a multi-selection menu and I had no problems with the check icons on the selected rows

![Image](https://github.com/user-attachments/assets/7856690d-c48b-4d75-a97a-cf7111bdbd47)
